crates/subspace-farmer/src/plot/tests.rs

```diff
@@ -132,3 +133,34 @@ async fn sequential_pieces_iterator() {
         .collect::<Vec<_>>();
     assert_eq!(got_indexes, piece_indexes[..got_indexes.len()]);
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn indexes_retrievable() {
+    init();
+    let base_directory = TempDir::new().unwrap();
+
+    let pieces = Arc::new(generate_random_pieces(10_000));
+    let offset = 0;
+
+    let plot = Plot::open_or_create(&base_directory, [0; 32].into(), u64::MAX).unwrap();
+    let count = pieces.count();
+    let piece_indexes = (offset..).take(count).collect::<Vec<_>>();
+    plot.write_many(Arc::clone(&pieces), piece_indexes.clone())
+        .unwrap();
+    let piece_indexes = (offset..)
+        .take(pieces.count())
+        .map(|index| (PieceIndexHash::from(index), index))
+        .collect::<BTreeMap<_, _>>();
+
+    for _ in 0..10000 {
+        let from = PieceIndexHash(rand::random());
+        let take = rand::thread_rng().gen_range(0..12000);
+        let indexes = plot.read_sequential_piece_indexes(from, take).unwrap();
+        let expected_indexes = piece_indexes
+            .range(from..)
+            .take(take as _)
+            .map(|(_, index)| *index)
+            .collect::<Vec<_>>();
+        assert_eq!(indexes, expected_indexes);
+    }
```

Comment:
This is really wasteful. Better write a few distinct cases that test edge-cases:
* non-wrapping simple case
* wrapping before reaching `max`
* wrapping that crosses `max`, but doesn't reach `min`
* wrapping that crosses `max` and reaches `min`

If those cases were all covered, you should have been able to see that current implementation isn't actually working correctly. WE don't need to run 10_000 random tests to check this (and spend a lot of resources. We also don't need to plot 10_000 pieces, something like 10 with correct position/public_key should be enough for this test.